client/state.cpp

```diff
@@ -227,36 +227,39 @@ std::vector<std::string> State::update(
   return upd;
 }
 
-bool State::update_frame(const torchcraft::fbs::FrameData* fd) {
-  if (fd->data() == nullptr)
-    return false;
-  if (fd->is_diff()) {
-    std::istringstream ss(std::string(
-        reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size()));
-    replayer::FrameDiff diff;
-    ss >> diff;
-    replayer::frame_undiff(this->frame, this->frame, &diff);
-  } else {
-    auto frame_string = std::string(
-        reinterpret_cast<const char*>(fd->data()->data()), fd->data()->size());
-    std::istringstream ss(frame_string);
-    ss >> *this->frame;
+bool State::update_frame(const void* flatBuffer, const torchcraft::fbs::FrameOrFrameDiff type) {
+  switch (type) {
+    case fbs::FrameOrFrameDiff::Frame: {
+      auto frameFlatBuffer = reinterpret_cast<const fbs::Frame*>(flatBuffer);
+      frame->readFromFlatBufferTable(*frameFlatBuffer);
+      return true;
+    }
+    case fbs::FrameOrFrameDiff::FrameDiff:  {
+      auto frameDiffFlatBuffer = reinterpret_cast<const fbs::FrameDiff*>(flatBuffer);
+      replayer::FrameDiff frameDiff;
+      frameDiff.readFromFlatBufferTable(*frameDiffFlatBuffer);
+      replayer::frame_undiff(frame, frame, &frameDiff);
+      return true;
```

Comment:
@jgehring This is the previously-discussed disuse of streams for deserialization